documentation/modules/con-external-authentication-service-api.adoc

```diff
@@ -0,0 +1,62 @@
+// Module included in the following assemblies:
+//
+// assembly-configuring.adoc
+
+[id='con-external-authentication-service-api-{context}']
+= External authentication service API
+
+An `external` authentication service must implement an AMQP SASL handshake, read the connection
+properties of the client, and respond with the expected connection properties containing the
+authentication and authorization information. The authentication service is queried by the address
+space components such as the router and broker, whenever a new connection is established to the
+messaging endpoints.
+
+== Authentication
+
+The requested identity of the client can be read from the SASL handshake `username`. The
+implementation can then authenticate the user.
+
+The authenticated identity should be returned in the `authenticated-identity` map with the following
+key/values (example uses JSON, but should be stored as an AMQP map on the connection property)
+
+[source,json,options="nowrap"]
+----
+{
+    "authenticated-identity": {
+        "sub: "myid",
+        "preferred_username": "myuser"
+    }
+}
+----
+
+== Authorization
+
+Authorization is a capability that can be requested by the client using the `ADDRESS-AUTHZ`
+connection capability. If this is set on the connection, the server should respond with this
+capability in the offered capabilities, and add the authorization information to the connection
+properties.
+
+The authorization information is contained within a map of address to list of operations allowed on
```

Comment:
...is stored within a map that correlates the address to a list of operations allowed on that address.